### PR TITLE
Restructuring version control chapter

### PR DESCRIPTION
**Pull request by christopherlovell**
# Summary
Addresses (but does not close)  #468.

### List of changes proposed in this PR (pull-request)

- Restructure Reproducible Environments chapter 
- Update TOCs

### What should a reviewer concentrate their feedback on?

- [ ] Everything looks ok?

### Acknowledging contributors

- [ ] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file: @christopherlovell

Thanks!

### CODE REVIEW

**Comment by KirstieJane** [comment]
Hi @allcontributors! Please could you add @christopherlovell for infrastructure!

**Comment by christopherlovell** [comment]
This PR currently breaks all the links in this chapter :grimacing: @KirstieJane and I spoke about the best approach to adding links between sections in chapters when split in to multiple files, and we weren't sure what the best approach was.

I've since found the following discussion on the jupyter-book repo:

https://github.com/jupyter/jupyter-book/issues/298

Essentially, it's tricky! If you put in hardcoded links to the markdown, these links break when compiled in to html. You can hard code the links to the `.html` compiled versions, but the question is do we want to also have the correct links to the original markdown?

File: book/content/version_control/commitmessages.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+## Commit messages
+
+### The problem
+
+As you work on you project you will make more and more commits.
+Without any other information it can be hard to remember which version of your project is in which.
+Storing past versions is useless if you ca not understand them, and figuring out what they contain by inspecting the code is frustrating and takes valuable time.
```
Comment (jbteves) [inline book/content/version_control/commitmessages.md:7]:
```suggestion
Storing past versions is useless if you can not understand them, and figuring out what they contain by inspecting the code is frustrating and takes valuable time.  
```

**Comment by jbteves** [comment]
Hello @christopherlovell! I'm not part of the Turing Way but I did participate on one of the collaboration cafes last month and chatted with @KirstieJane about the `git` chapter, and she pointed me to this PR. I corrected a small typo but didn't offer any other inline suggestions.

I'd like to add as an outsider that it might make sense to walk people through a little more global config setup, particularly handling the `autoclrf` problem that Windows users experience, and setting up the default editor. The editor gets a little funny if you use a GUI program like Notepad, because the program will continue with a save instead of with a quit since it launches outside of the terminal. So maybe some guidance there would be good, new users can be confused there.

I'm happy to open a PR to your fork to add those changes if you think they're worth discussing further.

**Comment by christopherlovell** [comment]
Hi @jbteves, thanks for taking a look! This PR was just for restructuring the files so that the content was not just in one single enormous file / page. I didn't have any involvement in editing the content, perhaps you could raise a PR for that separately?

Not sure if this PR is still relevant as I imagine there has been a lot of work on the content that these changes would significantly break. Probably worth closing this and attempting again....

**Comment by malvikasharan** [comment]
These changes have been integrated into the current version of the book. Thank you all for your contribution.

**Pull request closed without merging**